File: headless/src/main/org/nlogo/nvm/NonLocalExit.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.nvm;

// This is in Java for now because of
// issues.scala-lang.org/browse/SI-6063

public class NonLocalExit extends RuntimeException {
  // for efficiency, don't fill in stack trace
  @Override
  public RuntimeException fillInStackTrace() {
    return this;
  }
  public static final NonLocalExit singleton =
    new NonLocalExit();
}

File: netlogo-core/src/main/nvm/NonLocalExit.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.nvm

// this could be an object, but `catch { case NonLocalExit => ... }`
// will sometime change `StackOverflowError`s to `java.lang.NoClassDefFoundError`.
class NonLocalExit extends RuntimeException {
  // for efficiency, don't fill in stack trace
  override def fillInStackTrace = this
}
